lib/Service/SmimeService.php

```diff
@@ -517,8 +551,49 @@ public function isEncrypted(Horde_Imap_Client_Data_Fetch $message): bool {
 		/** @var Horde_Mime_Headers_ContentParam_ContentType $contentType */
 		$contentType = $headers['content-type'];
 		return $contentType->ptype === 'application'
-			&& $contentType->stype === 'pkcs7-mime'
+			&& ($contentType->stype === 'pkcs7-mime' || $contentType->stype === 'x-pkcs7-mime')
 			&& isset($contentType['smime-type'])
 			&& $contentType['smime-type'] === 'enveloped-data';
 	}
+
+	/**
+	 * Encrypt a MIME part using the given certificates.
+	 *
+	 * @param Horde_Mime_Part $part
+	 * @param SmimeCertificate[] $certificates
+	 * @return Horde_Mime_Part New MIME part containing the encrypted message and the signature
+	 *
+	 * @throws SmimeSignException If encrypting the message fails
```

Comment:
Technically not a `SignException` but we could generalize the exception to something like `SmimeException`.

Not really required for this PR though.

Reply:
> Not really required for this PR though.

Lazy me :see_no_evil:  I will add a proper exception ;)